nixos/modules/misc/documentation.nix

```diff
@@ -51,12 +76,11 @@ let
           (name: value:
             let
               wholeName = "${namePrefix}.${name}";
-              guard = lib.warn "Attempt to evaluate package ${wholeName} in option documentation; this is not supported and will eventually be an error. Use `mkPackageOption{,MD}` or `literalExpression` instead.";
+              guard = warn "Attempt to evaluate package ${wholeName} in option documentation; this is not supported and will eventually be an error. Use `mkPackageOption{,MD}` or `literalExpression` instead.";
             in if isAttrs value then
               scrubDerivations wholeName value
               // optionalAttrs (isDerivation value) {
                 outPath = guard "\${${wholeName}}";
-                drvPath = guard drvPath;
```

Comment:
Why was this line removed?

Reply:
It doesn't eval; there's no drvPath in scope. It was masked by the top-level with.

Reply:
I noted this in the PR head matter.

Reply:
Ah, it all makes sense now. I didn't understand that line when I first read it in the description.

Reply:
I'd be glad to, but I don't understand how it's a bug if the code wasn't ever evaluated in the first place. Doesn't that just meet the definition of dead code?

Definitely this code meant to do something, and I'd be happy to help make it do whatever it was meant to do.

Reply:
I was writing the suggestion! :) Couldn't suggest a fix because GH doesn't like suggestion on removed lines. Here it is: https://github.com/NixOS/nixpkgs/pull/299650#discussion_r1546992510

Reply:
To answer the question here since it got asked here: this is not dead code because the purpose of the code is to ensure that it will not be used - hence the `lib.warn` line. It exists to let people know what to do in case they eventually hit this code by accident. If the code isn't ever evaluated, it's doing its job.